mypy/binder.py

```diff
@@ -439,8 +457,13 @@ def top_frame_context(self) -> Iterator[Frame]:
 
 
 def get_declaration(expr: BindableExpression) -> Type | None:
-    if isinstance(expr, RefExpr) and isinstance(expr.node, Var):
-        type = expr.node.type
-        if not isinstance(get_proper_type(type), PartialType):
-            return type
+    if isinstance(expr, RefExpr):
+        if isinstance(expr.node, Var):
+            type = expr.node.type
+            if not isinstance(get_proper_type(type), PartialType):
+                return type
+        elif isinstance(expr.node, TypeInfo):
+            return Instance(
```

Comment:
I'm not sure this is right... The declared type of a TypeInfo should probably be `expr.node.calculate_metaclass_type()` right?

I don't have the best sense of this code either, so not sure how much of a difference this makes though.

I did notice this issue while reviewing this PR: https://github.com/python/mypy/pull/13483